src/main/java/de/tum/cit/aet/artemis/assessment/service/ResultService.java

```diff
@@ -530,31 +535,66 @@ private Result shouldSaveResult(@NotNull Result result, boolean shouldSave) {
+        feedbackDetails = feedbackDetails.stream().map(detail -> {
             double relativeCount = (detail.count() * 100.0) / distinctResultCount;
-            int taskNumber = tasks.stream().filter(task -> task.getTestCases().stream().anyMatch(tc -> tc.getTestName().equals(detail.testCaseName()))).findFirst()
-                    .map(task -> tasks.stream().toList().indexOf(task) + 1).orElse(0);
+            int taskNumber = determineTaskNumberOfTestCase(detail.testCaseName(), tasks);
+
             return new FeedbackDetailDTO(detail.count(), relativeCount, detail.detailText(), detail.testCaseName(), taskNumber);
-        }).toList();
+        }).filter(matchesSearchTerm(searchTerm)).sorted(getComparatorForFeedbackDetails(search)).toList();
+
+        return paginateFeedbackDetails(feedbackDetails, search.getPage(), search.getPageSize());
+    }
+
+    private Predicate<FeedbackDetailDTO> matchesSearchTerm(String searchTerm) {
+        return detail -> searchTerm.isEmpty() || detail.detailText().toLowerCase().contains(searchTerm) || detail.testCaseName().toLowerCase().contains(searchTerm)
+                || String.valueOf(detail.count()).contains(searchTerm) || String.valueOf(detail.taskNumber()).contains(searchTerm)
+                || String.valueOf(detail.relativeCount()).contains(searchTerm);
+    }
+
+    private Comparator<FeedbackDetailDTO> getComparatorForFeedbackDetails(SearchTermPageableSearchDTO<String> search) {
+        Map<String, Comparator<FeedbackDetailDTO>> comparators = Map.of("count", Comparator.comparingLong(FeedbackDetailDTO::count), "detailText",
+                Comparator.comparing(FeedbackDetailDTO::detailText, String.CASE_INSENSITIVE_ORDER), "testCaseName",
+                Comparator.comparing(FeedbackDetailDTO::testCaseName, String.CASE_INSENSITIVE_ORDER), "taskNumber", Comparator.comparingInt(FeedbackDetailDTO::taskNumber),
+                "relativeCount", Comparator.comparingDouble(FeedbackDetailDTO::relativeCount));
+        Comparator<FeedbackDetailDTO> comparator = comparators.getOrDefault(search.getSortedColumn(), (a, b) -> 0);
+        return search.getSortingOrder() == SortingOrder.ASCENDING ? comparator : comparator.reversed();
+    }
+
+    private int determineTaskNumberOfTestCase(String testCaseName, List<ProgrammingExerciseTask> tasks) {
+        return tasks.stream().filter(task -> task.getTestCases().stream().anyMatch(tc -> tc.getTestName().equals(testCaseName))).findFirst()
+                .map(task -> tasks.stream().toList().indexOf(task) + 1).orElse(0);
+    }
+
+    private FeedbackAnalysisResponseDTO paginateFeedbackDetails(List<FeedbackDetailDTO> feedbackDetails, int page, int pageSize) {
+        int start = (page - 1) * pageSize;
+        int end = Math.min(start + pageSize, feedbackDetails.size());
+        List<FeedbackDetailDTO> paginatedFeedbackDetails = feedbackDetails.subList(start, end);
+
+        int totalPages = (feedbackDetails.size() + pageSize - 1) / pageSize;
```

Comment:
Possible follow-up: This approach works, but is quite unintuitive hand hard to understand. Consider using Math.ceil instead.